pkg/manifests/config.go

```diff
@@ -207,6 +208,10 @@ func NewConfig(content io.Reader, tp bool) (*Config, error) {
 		return nil, errors.Wrap(ErrConfigValidation, "collectionProfiles is a TechPreview feature, to be able to use a profile different from the default (\"full\") please enable TechPreview")
 	}
 
+	if c.ClusterMonitoringConfiguration.MetricsServerConfig != nil && !metricsServerEnabled {
```

Comment:
since we have a dedicated feature gate for metrics server, I wouldn't error out if the config is present. It can even be preferred to handle upgrades (e.g. you can setup the metrics server config before it goes GA).